%YAML:1.0
cameras:
     - {T_SC:
        [ 0, 0, 1, -0.02,
          1, 0, 0, 0,
           0, 1, 0, 0.06,
           0.0, 0.0, 0.0, 1.0],
        image_dimension: [640, 480],
        distortion_coefficients: [0, 0, 0, 0],
        distortion_type: radialtangential,
        focal_length: [615, 615],
        principal_point: [320, 240],
        image_readout_time: 0.00004}

camera_params:
    camera_rate: 10 # just to manage the expectations of when there should be frames arriving
    sigma_absolute_translation: 0.00005 # The standard deviation of the camera extrinsics translation, e.g. 1.0e-10 for online-calib [m].
    sigma_absolute_orientation: 0.0 # The standard deviation of the camera extrinsics orientation, e.g. 1.0e-3 for online-calib [rad].
    sigma_c_relative_translation: 0.0 # The std. dev. of the cam. extr. transl. change between frames, e.g. 1.0e-6 for adaptive online calib (not less for numerics) [m].
    sigma_c_relative_orientation: 0.0 # The std. dev. of the cam. extr. orient. change between frames, e.g. 1.0e-6 for adaptive online calib (not less for numerics) [rad].
    timestamp_tolerance: 0.005 #[s]stereo frame out-of-sync tolerance
    sigma_focal_length: 0.1
    sigma_principal_point: 0.1
    sigma_distortion: [0.001, 0.0001, 0.0001, 0.0001, 0.00001]
    sigma_td: 0.00001
    sigma_tr: 0.000005

imu_params:
    a_max: 176.0 # acceleration saturation [m/s^2]
    g_max: 7.8 # gyro saturation [rad/s]
    sigma_g_c: 5.24e-4 # gyro noise density [rad/s/sqrt(Hz)] 0.03 deg/sec/sqrt(Hz), following Microstrain 3dm gx3 -35, the noisy samples are drawn with noise half of these stds of n_g, n_a, n_bg and n_ba
    sigma_a_c: 8.0e-4 # accelerometer noise density [m/s^2/sqrt(Hz)] 80 ug
    sigma_bg: 0.03 # gyro bias prior [rad/s]
    sigma_ba: 0.1 # accelerometer bias prior [m/s^2]
    sigma_gw_c: 1.45e-6 # gyro drift noise density [rad/s^s/sqrt(Hz)] 18 deg/hr in one hour
    sigma_aw_c: 6.7e-6 # accelerometer drift noise density [m/s^3/sqrt(Hz)] 4e-4 m/s^2 in one hour
    tau: 3600.0 # reversion time constant, currently not in use [s]
    g: 9.81007 # Earth's acceleration due to gravity [m/s^2]
    a0: [ 0.0, 0.0, 0.0 ] # Accelerometer bias [m/s^2]
    imu_rate: 100
    # tranform Body-Sensor (IMU)
    T_BS:
        [1.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 1.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 1.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 1.0000]
    sigma_TGElement: 5e-4
    sigma_TSElement: 5e-4
    sigma_TAElement: 5e-4

# Estimator parameters
numKeyframes: 5 # number of keyframes in optimisation window
numImuFrames: 3 # number of frames linked by most recent nonlinear IMU error terms

# ceres optimization options
ceres_options:
    minIterations: 3   # minimum number of iterations always performed
    maxIterations: 10  # never do more than these, even if not converged
    timeLimit: 0.035   # [s] negative values will set the an unlimited time limit

# detection
detection_options:
    threshold: 30.0      # detection threshold. By default the uniformity radius in pixels
    octaves: 2           # number of octaves for detection. 0 means single-scale at highest resolution
    maxNoKeypoints: 500  # restrict to a maximum of this many keypoints per image (strongest ones)

# delay of images [s]:
imageDelay: 0.0  # in case you are using a custom setup, you will have to calibrate this. 0 for the VISensor.

# display debug images?
displayImages: true  # displays debug video and keyframe matches. May be slow.

# use direct driver
useDriver: false 

# some options for how and what to publish -- optional in ROS-free version
publishing_options:
    publish_rate: 25                  # rate at which odometry updates are published only works properly if imu_rate/publish_rate is an integer!!
    publishLandmarks: true             # select, if you want to publish landmarks at all
    landmarkQualityThreshold: 1.0e-2   # landmark with lower quality will not be published
    maximumLandmarkQuality: 0.05       # landmark with higher quality will be published with the maximum colour intensity
    maxPathLength: 20                  # maximum length of the published path
    publishImuPropagatedState: false   # Should the state that is propagated with IMU messages be published? Or just the optimized ones?
    # provide custom World frame Wc
    T_Wc_W:
        [1.0000, 0.0000, 0.0000, 0.0000,
         0.0000, 1.0000, 0.0000, 0.0000,
         0.0000, 0.0000, 1.0000, 0.0000,
         0.0000, 0.0000, 0.0000, 1.0000]
    trackedBodyFrame: B                # B or S, the frame of reference that will be expressed relative to the selected worldFrame
    velocitiesFrame: Wc                # Wc, B or S,  the frames in which the velocities of the selected trackedBodyFrame will be expressed in
    outputPath: "/home/jhuai/Desktop"

input_data:
    image_folder: "/media/jhuai/Mag/NewTsukubaStereoDataset/illumination/daylight"
    time_file: "/media/jhuai/Mag/NewTsukubaStereoDataset/SimulateIMUTest/times.txt"
    imu_file: "/media/jhuai/Mag/NewTsukubaStereoDataset/SimulateIMUTest/samplesnoisy_timeoffset.txt"
    startIndex:0
    finishIndex:1799

initial_state:
    v_WS: [0.00,0.00,0.00]
    std_v_WS:
	[0.01, 0.01, 0.01]
